test/functorch/test_control_flow.py

```diff
@@ -2011,13 +2117,196 @@ def chain_fct_different_dim(inp):
+            return x * cnt
+
+        scan_fct = compile_mode_helper(associative_scan, compile_mode)
+
+        inp = torch.randn(3, 10, 1, device=device) * 2
+        
+        result = scan_fct(loop_body, inp, dim=0, reverse=reverse, combine_mode=combine_mode)
+        expected_result = _fake_associative_scan(
+            loop_body,
+            inp,
+            0,
             reverse=reverse,
         )
-        result1 = fct_cmp(inp)
-        self.assertEqual(result1, expected_result)
+        
+        self.assertEqual(result, expected_result)
+        
+    # TODO: provide an implementation for all compile modes and re-enable all test
+    @unittest.skipIf(not SM70OrLater, "triton")
+    @requires_cuda
+    @parametrize("combine_mode", ["pointwise", "generic"])
+    @parametrize("compile_mode", ["none", "compile", "compile_dynamic_shape"])
+    @parametrize("reverse", [False, True])
+    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    # Skipping combine_mode=pointwise 
+    # as the cond operation is classified to be non-pointwise
+    @decorateIf(
+        unittest.skip,
+        lambda params: (
+            params["combine_mode"] == "pointwise"
+            # and (params["device"] == torch.device("cpu") or torch.version.hip)
+        ),
+    )
+    def test_associative_scan_cond(
+        self, combine_mode, compile_mode, reverse, device
+    ):
+        def body(x, y):
+            # TODO: Is the cond operation really non-pointwise?
+            val = cond(torch.sum(y) > 0., lambda y: y + 0., lambda y: 1. - y, (y,))
+            return x * val
+
+        scan_fct = compile_mode_helper(associative_scan, compile_mode)
+
+        inp = torch.randn(3, 10, 1, device=device)
+        
+        result = scan_fct(body, inp, dim=0, reverse=reverse, combine_mode=combine_mode)
+        expected_result = _fake_associative_scan(
+            body,
+            inp,
+            0,
+            reverse=reverse,
+        )
+        
+        self.assertEqual(result, expected_result)
+        
+    # TODO: provide an implementation for all compile modes and re-enable all test
+    @unittest.skipIf(not SM70OrLater, "triton")
+    @requires_cuda
+    @parametrize("combine_mode", ["pointwise", "generic"])
+    @parametrize("compile_mode_scan", ["none", "compile", "compile_dynamic_shape"])
+    @parametrize("compile_mode_associative_scan", ["none", "compile", "compile_dynamic_shape"])
+    @parametrize("reverse", [False, True])
+    @parametrize("reverse_associative_scan", [False, True])
+    @parametrize("device", [torch.device("cpu"), torch.device("cuda")])
+    # Skipping combine_mode=pointwise 
+    # as the cond operation is classified to be non-pointwise
+    @decorateIf(
+        unittest.skip,
+        lambda params: (
+            params["combine_mode"] == "pointwise"
+            # and (params["device"] == torch.device("cpu") or torch.version.hip)
+        ),
+    )
+    def test_scan_associative_scan(
```

Comment:
the test name should be something like test_associative_scan_in_scan or something like that, right?

Can we do the same for all the nesting tests?

Reply:
Done, I renamed the tests